packages/web-react/src/components/Button/useButtonAriaProps.ts

```diff
@@ -1,60 +1,60 @@
-import { AnchorHTMLAttributes, ButtonHTMLAttributes, ElementType, HTMLAttributes } from 'react';
-import { ClickEvent, AriaButtonProps } from '../../types';
-
-export interface ButtonAria<T> {
-  /** Props for the button element. */
-  buttonProps: T;
-}
-
-export function useButtonAriaProps<C = void, S = void>(
-  props: AriaButtonProps<'a', C, S>,
-): ButtonAria<AnchorHTMLAttributes<HTMLAnchorElement>>;
-export function useButtonAriaProps<C = void, S = void>(
-  props: AriaButtonProps<'button', C, S>,
-): ButtonAria<ButtonHTMLAttributes<HTMLButtonElement>>;
-export function useButtonAriaProps<C = void, S = void>(
-  props: AriaButtonProps<ElementType, C, S>,
-): ButtonAria<HTMLAttributes<HTMLElement>>;
-
-export function useButtonAriaProps<C = void, S = void>(
-  props: AriaButtonProps<ElementType, C, S>,
-): ButtonAria<HTMLAttributes<unknown>> {
-  const { elementType = 'button', isDisabled, onClick, href, target, rel, type = 'button', ariaLabel } = props;
-
-  let additionalProps;
-  if (elementType === 'button') {
-    additionalProps = {
-      type,
-      disabled: isDisabled,
-    };
-  } else {
-    additionalProps = {
-      role: 'button',
-      href: elementType === 'a' && isDisabled ? undefined : href,
-      target: elementType === 'a' ? target : undefined,
-      type: elementType === 'a' && type === 'button' ? undefined : type,
-      disabled: isDisabled,
-      rel: elementType === 'a' ? rel : undefined,
-    };
+import { ClickEvent, SpiritButtonProps, SpiritButtonLinkProps } from '../../types';
+
+const handleClick = (event: ClickEvent, isDisabled?: boolean, onClick?: (event: ClickEvent) => void) => {
+  if (isDisabled) {
+    event.preventDefault();
+
+    return;
+  }
+
+  if (onClick) {
+    onClick(event);
   }
+};
 
-  const handleClick = (event: ClickEvent) => {
-    if (isDisabled) {
-      event.preventDefault();
+export type UseButtonAriaProps = Partial<SpiritButtonProps>;
+export type UseButtonAriaReturn = {
+  buttonProps: Partial<SpiritButtonProps>;
+};
 
-      return;
-    }
+export const useButtonAriaProps = (props: UseButtonAriaProps): UseButtonAriaReturn => {
+  const { isDisabled, onClick, type = 'button', ariaLabel } = props;
 
-    if (onClick) {
-      onClick(event);
-    }
+  const additionalProps = {
+    type,
+    disabled: isDisabled,
   };
 
   return {
     buttonProps: {
       ...additionalProps,
-      onClick: handleClick,
+      onClick: (event) => handleClick(event, isDisabled, onClick),
+      'aria-label': ariaLabel || undefined,
+    },
+  };
+};
+
+export type UseButtonLinkAriaProps = Partial<SpiritButtonLinkProps>;
+export type UseButtonLinkAriaReturn = {
+  buttonLinkProps: Partial<SpiritButtonLinkProps>;
```

Comment:
Why is this definition doubled here? E.g. the Partial has been defined as a line above.

Reply:
Because there are two hooks, `useButtonAriaProps` and `useButtonLinkAriaProps`, and each of them has its own type. For better clarity, I have defined both prop types, even though they are similar

Reply:
So I can set `buttonLinkProps: UseButtonLinkAriaProps;` if this is what you mean

Reply:
> So I can set `buttonLinkProps: UseButtonLinkAriaProps;` if this is what you mean

Yeah, that is what I mean. Because it makes no sense to me to repeat a type when I have the definition a line above.  :-)